doc/developers/auth/jwt.rst

```diff
@@ -0,0 +1,139 @@
+JWT Authentication
+==================
+
+JWT (`JSON Web Token <https://tools.ietf.org/html/rfc7519>`_) bearer authentication allows to use an external identity provider (IdP) such as the `Microsoft Identity Platform <https://aka.ms/IdentityPlatform>`_ for user authentication in CCF.
+
+Once the user has acquired a token from an IdP supported by the app, they can include it in HTTP requests in the ``Authorization`` header as `bearer token <https://tools.ietf.org/html/rfc6750>`_.
+The CCF app validates the token and can then use the user identity and other claims embedded in the token.
+
+CCF provides support for storing and retrieving public token signing keys required for validating tokens.
+In the future, CCF will support validating token signatures (currently the responsibility of apps) and automatically refreshing token signing keys from OpenID Provider Configuration endpoints.
+
+Storing public token signing keys
+---------------------------------
+
+IdPs sign tokens with private keys that are periodically updated.
+Corresponding public keys in the form of certificates are needed to validate the signature of a token and are typically published by IdPs at a well-known location in JWKS (`JSON Web Key Set <https://tools.ietf.org/html/rfc7517>`_) format.
+
+.. note::
+
+    Most IdPs support the OpenID Connect Discovery specification which defines how to discover the location to the  certificates.
+    In short, the IdP issuer URL, for example ``https://login.microsoftonline.com/common/v2.0``, is suffixed with ``/.well-known/openid-configuration`` and the ``jwks_uri`` field of the JSON document at that URL is the location of the JWKS     document containing the certificates currently in use. In this example, it would be `<https://login.microsoftonline.com/    common/discovery/v2.0/keys>`_.
```

Comment:
```suggestion
    In short, the IdP issuer URL, for example ``https://login.microsoftonline.com/common/v2.0``, is suffixed with ``/.well-known/openid-configuration`` and the ``jwks_uri`` field of the JSON document at that URL is the location of the JWKS document containing the certificates currently in use. In this example, it would be `<https://login.microsoftonline.com/common/discovery/v2.0/keys>`_.
```